### PR TITLE
Upgrade sqlalchemy to version 2

### LINKED ISSUE #489: Upgrade to sqlalchemy 2
SqlAlchemy query is a bit outdated and switching to SqlAlchemy 2 would allow use to use the replacement style in the form of select

### PR DESCRIPTION
**Pull request by emanuelen5**
Closes #489 

## Summary by CodeRabbit

- **New Features**
	- Enhanced SQL query handling by utilizing the `text()` construct for improved clarity and maintainability.

- **Bug Fixes**
	- Streamlined member retrieval across various functions, improving efficiency and readability.

- **Chores**
	- Updated dependency version for `sqlalchemy` to ensure compatibility with version 2.0.
	- Removed deprecated warning filters from configuration files.

### CODE REVIEW

**Comment by emanuelen5** [comment]
The tests still fail. Simple reproducer:

```bash
PYTEST_ADDOPTS='-k test_reminder_message_is_created_20_days_before_expiry_even_if_other_span_after' make test
```

**Comment by emanuelen5** [comment]
@coderabbitai review

File: api/src/member/member.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def send_access_token_email(redirect, user_identification, ip, browser):
 
 
 def send_updated_member_info_email(member_id: int, msg_swe: str, msg_en: str):
-    member = db_session.query(Member).get(member_id)
+    member = db_session.get(Member, member_id)
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/member/member.py:40]: Fixed

File: api/src/shop/ordered_entity.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def create(self, data=None, commit=True):
         if data is None:
             data = request.json or {}
 
-        (status,) = db_session.execute("SELECT GET_LOCK('display_order', 20)").fetchone()
+        (status,) = db_session.execute(text("SELECT GET_LOCK('display_order', 20)")).fetchone()
         if not status:
             raise InternalServerError("Failed to create, try again later.", log="failed to aquire display_order lock")
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/shop/ordered_entity.py:22]: Won't fix any of the text statements that you suggested. That will have to be done in a later PR. This PR just addresses the compatibility with SQL without making any more behavioral change.

File: api/src/init_db.py [hunk 3]
```diff
@@ -11,14 +11,15 @@
...
+from sqlalchemy import text
 from sqlalchemy.orm import sessionmaker
 from sqlalchemy.orm.exc import MultipleResultsFound, NoResultFound
 
 
 def clear_permission_cache(session_factory):
     """Clear permisssion cache as a part of every db_init/restart."""
     with closing(session_factory()) as session:
-        session.execute("UPDATE access_tokens SET permissions = NULL")
+        session.execute(text("UPDATE access_tokens SET permissions = NULL"))
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/init_db.py:22]: Won't change the exception-handling that isn't related to the SQLalchemy 2.0 upgrade

File: api/src/core/models.py [hunk 4]
```diff
@@ -46,19 +45,21 @@ class Login:
...
 
     @staticmethod
     def get_failed_login_count(ip):
         (count,) = db_session.execute(
-            "SELECT count(1) FROM login"
-            "   WHERE ip = :ip AND NOT success AND date >= DATE_SUB(NOW(), INTERVAL 1 HOUR)",
+            text(
+                "SELECT count(1) FROM login"
+                "   WHERE ip = :ip AND NOT success AND date >= DATE_SUB(NOW(), INTERVAL 1 HOUR)"
+            ),
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/core/models.py:62]: No, the space is there

File: api/src/membership/member_entity.py [hunk 5]
```diff
@@ -46,7 +47,7 @@ def create(self, data=None, commit=True):
                 with db_session.begin_nested():
                     data = data.copy()
                     (max_member_number,) = db_session.execute(
-                        "SELECT COALESCE(MAX(member_number), 999) FROM membership_members"
+                        text("SELECT COALESCE(MAX(member_number), 999) FROM membership_members")
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/membership/member_entity.py:50]: Not related to sqlalchemy 2.0 won't fix in this PR

File: api/src/shop/shop_data.py [hunk 6]
```diff
@@ -76,7 +76,7 @@ def receipt(member_id, transaction_id):
         transaction = (
             db_session.query(Transaction)
             .filter_by(member_id=member_id, id=transaction_id)
-            .options(joinedload("contents"), joinedload("contents.product"))
+            .options(joinedload(Transaction.contents).joinedload(TransactionContent.product))
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/shop/shop_data.py:79]: That won't help readability

File: api/src/migrate.py [hunk 7]
```diff
@@ -115,10 +133,10 @@ def run_migrations(session_factory):
...
 
                 for sql in read_sql(join(migrations_dir, migration.name + ".sql")):
-                    session.execute(sql)
+                    session.execute(text(sql))
 
                 session.execute(
-                    "INSERT INTO migrations VALUES (:id, :name, :applied_at)",
+                    text("INSERT INTO migrations VALUES (:id, :name, :applied_at)"),
                     {"id": migration.id, "name": migration.name, "applied_at": datetime.utcnow()},
                 )
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/migrate.py:141]: Not related to sqlalchemy 2.0 upgrade. Won't fix in this PR.

File: api/src/migrate.py [hunk 8]
```diff
@@ -26,53 +26,71 @@ def ensure_migrations_table(engine, session_factory):
...
-                "    WHERE id=2 AND service='core' AND name='0002_password_reset_token'"
+                text(
+                    "UPDATE migrations SET id=8, name='0008_password_reset_token'"
+                    "    WHERE id=2 AND service='core' AND name='0002_password_reset_token'"
+                )
             )
-            session.execute("ALTER TABLE migrations DROP PRIMARY KEY, ADD PRIMARY KEY(id)")
-            session.execute("ALTER TABLE migrations DROP COLUMN service")
+            session.execute(text("ALTER TABLE migrations DROP PRIMARY KEY, ADD PRIMARY KEY(id)"))
+            session.execute(text("ALTER TABLE migrations DROP COLUMN service"))
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/migrate.py:93]: Not related to sqlalchemy 2.0 upgrade. Won't fix in this PR.

File: api/src/core/auth.py [hunk 9]
```diff
@@ -131,7 +131,7 @@ def password_reset(reset_token, unhashed_password):
         raise BadRequest(str(e))
 
     try:
-        member = db_session.query(Member).get(password_reset_token.member_id)
+        member = db_session.get(Member, password_reset_token.member_id)
     except NoResultFound:
         raise InternalServerError(log=f"No member with id {password_reset_token.member_id} found, this is a bug.")
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/core/auth.py]: Fixed

File: api/src/migrate.py [hunk 10]
```diff
@@ -26,53 +26,71 @@ def ensure_migrations_table(engine, session_factory):
...
-                "    PRIMARY KEY (id)"
-                ") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci"
+                text(
+                    "CREATE TABLE migrations ("
+                    "    id INTEGER NOT NULL,"
+                    "    name VARCHAR(255) COLLATE utf8mb4_0900_ai_ci NOT NULL,"
+                    "    applied_at DATETIME NOT NULL,"
+                    "    PRIMARY KEY (id)"
+                    ") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci"
+                )
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/migrate.py:38]: Not related to sqlalchemy 2.0 upgrade. Won't fix in this PR.

File: api/src/shop/stripe_payment_intent.py [hunk 11]
```diff
@@ -159,7 +159,7 @@ def pay_with_stripe(
     """Handle stripe payment"""
 
     try:
-        member = db_session.query(Member).get(transaction.member_id)
+        member = db_session.get(Member, transaction.member_id)
         assert member is not None
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply api/src/shop/stripe_payment_intent.py:163]: Not related to sqlalchemy 2.0 upgrade. Won't fix in this PR.

**Comment by emanuelen5** [comment]
@coderabbitai review

**Review by BerglundDaniel — APPROVED** [review]

**Pull request merged**